shared_model/validators/field_validator.cpp

```diff
@@ -106,10 +106,10 @@ namespace shared_model {
 
     void FieldValidator::validateAmount(ReasonsGroupType &reason,
                                         const interface::Amount &amount) const {
-      if (amount.intValue() <= 0) {
+      if (amount.sign() <= 0) {
         auto message =
-            (boost::format("Amount must be greater than 0, passed value: %d")
-             % amount.intValue())
+            (boost::format("Amount must be greater than 0, but got: %d")
+             % amount.toStringRepr())
```

Comment:
It will output `NaN`, because the string repr is only initialized with positive numbers.

Reply:
Iroha API allows operating only with non-negative numbers. Thus, I do not see the issue.

Reply:
the issue is logs reading "but got: NaN", which is misleading.

Reply:
and '%d' in format is an issue too